synapse/cores/common.py

```diff
@@ -1938,3 +1912,68 @@ def _tufosByInetCidr(self, prop, valu, limit=None):
 
         return self.getTufosBy('range', prop, (ipv4addr, ipv4addr+mask), limit=limit)
 
+    #############################################################
+    # support typelib persistence
+    #############################################################
+
+    # overrides: synapse.lib.types.TypeLib.addType
+    def addType(self, typ):
+        s_types.TypeLib.addType(self, typ)
+        self.formTufoByFrob('syn:type', typ.name)
+
+    # overrides: synapse.lib.types.TypeLib.addSubType
+    def addSubType(self, name, subof, **info):
+        s_types.TypeLib.addSubType(self, name, subof, **info)
+        base = self.getTufoByFrob('syn:type', subof)
+
+        subinfo = {}
+        for propname, propval in base[1].items():
+            if not propname.startswith('syn:type:'):
+                continue
+            propname = propname.partition('syn:type')[2]
+            subinfo[propname] = propval
+
+        for propname, propval in info.items():
+            subinfo[propname] = propval
+
+        subinfo['base'] = subof
+
+        if 'fields' in subinfo:
+            fields = subinfo.pop('fields')
+            subinfo['fields'] = len(fields)
+            for i, (fieldname, fieldtype) in enumerate(fields):
+                k = 'field:%d:%s' % (i, fieldname)
+                subinfo[k] = fieldtype
```

Comment:
Wondered how you were gonna handle that one.  It's the nastiest example of this issue by far. The only other option might be to create a spec for flattening json data into tufo-per-field representations with a secondary prop to lift them all at once. Any other crazy/bright ideas for dealing with this in a more generalized way?